app/controllers/concerns/secondary_authentication_concern.rb

```diff
@@ -12,7 +12,7 @@ def require_secondary_authentication(redirect_to: request.fullpath)
 
     if user_id_for_secondary_authentication && !secondary_authentication_present?
       user = User.find(user_id_for_secondary_authentication)
-      session[:secondary_authentication_redirect_to] = redirect_to
+      session[:secondary_authentication_redirect_to] = request.get? ? redirect_to : "/"
```

Comment:
```suggestion
      if request.get?
        session[:secondary_authentication_redirect_to] = redirect_to
        session[:secondary_authentication_redirect_to_post_blocked] = false
      else
        session[:secondary_authentication_redirect_to_post_blocked] = true
      end
```
This would mean that if a POST request is sent the redirect would remain the last GET endpoint they accessed (which would be the form).

The `secondary_authentication_redirect_to_post_blocked` flag could be used to render a flash message when the user is redirected to the form informing them that they have to try again.